apps/core/scenes/search/PlacePicker/PlacePickerContent.js

```diff
@@ -2,117 +2,133 @@
 
-import {
-  graphql,
-  createRefetchContainer,
-  type RelayRefetchProp,
-} from '@kiwicom/margarita-relay';
 import * as React from 'react';
-import { View, ScrollView } from 'react-native';
+import { View } from 'react-native';
 import {
   Text,
   TextInput,
   Icon,
   StyleSheet,
 } from '@kiwicom/universal-components';
 
-import type { PlacePickerContent_locations as PlacePickerContentType } from './__generated__/PlacePickerContent_locations.graphql.js';
+import { MODAL_TYPE } from '../SearchConstants';
+import {
+  withSearchContext,
+  type SearchContextState,
+  type Location,
+} from '../SearchContext';
+import PlacePickerList from './PlacePickerList';
+
+type Locations = $ReadOnlyArray<?{|
+  +node: ?{|
+    +id: string,
+    +name: ?string,
+    +locationId: ?string,
+  |},
+|}>;
 
 type Props = {|
-  +locations: ?PlacePickerContentType,
-  +type: string,
-  +relay: RelayRefetchProp,
+  +locations: Locations,
+  +onChangeText: string => void,
+  +travelFrom: ?Location,
+  +travelTo: ?Location,
+  +type: 'DESTINATION' | 'ORIGIN',
+  +setTravelTo: Location => void,
+  +setTravelFrom: Location => void,
+  +setModalType: string => void,
 |};
 
 type State = {|
   text: string,
 |};
 
-class PlacePicker extends React.Component<Props, State> {
-  state = {
-    text: '',
-  };
+const setDefaultValue = (pickerType: string, from: string, to: string) => {
+  if (pickerType === MODAL_TYPE.ORIGIN) return from;
+  if (pickerType === MODAL_TYPE.DESTINATION) return to;
+  return '';
+};
 
-  handleChangeText = (text: string) => {
-    this.setState({ text });
-    this.props.relay.refetch({ input: { term: text } });
-  };
+class PlacePickerContent extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      text: setDefaultValue(
+        props.type,
+        props.travelFrom?.name,
+        props.travelTo?.name,
+      ),
+    };
+  }
 
   getLabel = () => {
     const { type } = this.props;
-    if (type === 'DESTINATION') {
+
+    if (type === MODAL_TYPE.DESTINATION) {
       return 'To';
-    } else if (type === 'ORIGIN') {
+    } else if (type === MODAL_TYPE.ORIGIN) {
       return 'From';
     }
     return null;
   };
 
-  render() {
-    const locations = this.props.locations?.locationsByTerm?.edges ?? [];
+  handleChangeText = (text: string) => {
+    this.setState({ text });
+    this.props.onChangeText(text);
+  };
+
+  handleListItemClick = location => {
+    const { type, setTravelFrom, setTravelTo, setModalType } = this.props;
 
+    if (type === MODAL_TYPE.ORIGIN) {
+      setTravelFrom(location);
+    }
+    if (type === MODAL_TYPE.DESTINATION) {
+      setTravelTo(location);
+    }
+    setModalType(MODAL_TYPE.HIDDEN);
+  };
+
+  render() {
     return (
       <View style={styles.container}>
         <View>
           <Text weight="bold">{this.getLabel()}:</Text>
           <TextInput
+            autofoucs
```

Comment:
`autofoucs` -> `autofocus`

Reply:
🙈

Reply:
But currently, there is no feature for this in the TextInput component. So I will delete it. I created the issue in `universal-components` for that. https://github.com/kiwicom/universal-components/issues/238